### PR TITLE
Add nearby amenity search for rerouting

### PR DESCRIPTION
**Pull request by ddohler**
This is basically a thin wrapper around Google's Places Javascript API. Lots of TODOs here; this is just a stub to get the communication with Google working and output it to the console.

### CODE REVIEW

File: src/nih_wayfinding/app/scripts/amenities/amenities-search-service.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        var module = {
+            searchNearby: searchNearby
+        };
+
+        return module;
+
+        // Internal function implementations
+        /**
+         * Find nearby amenities to which the user can be re-routed
+         * @param radius {list} Array containing point coordinates in [lat, lng] order
```
Comment (CloudNiner) [inline src/nih_wayfinding/app/scripts/amenities/amenities-search-service.js]: radius -> point

File: src/nih_wayfinding/app/scripts/routing/rerouting/rerouting-options-service.js [hunk 2]
```diff
@@ -2,16 +2,42 @@
...
         return module;
 
         function getReroutingOptions() {
             return ReroutingOptions;
         }
+
+        function reroute(preferences) {
+            // TODO: Update this when location mocking is in place
+            var currentLocation = [41.72866461875987,-87.56206512451172];
+            // TODO: This probably should be determined somewhere else
```
Comment (CloudNiner) [inline src/nih_wayfinding/app/scripts/routing/rerouting/rerouting-options-service.js]: Could just set this in the global config.js

**Comment by CloudNiner** [comment]
+1

**Comment by moradology** [comment]
Before you merge, be sure to add
```javascript
'app/scripts/amenities/module.js',
'app/scripts/amenities/amenities-search-service.js',
```
in karma.conf.js. Tests fail otherwise.

**Comment by moradology** [comment]
Merge in the changes I made a PR for and then this'll be good to go

**Pull request merged**